src/x509_extensions.rs

```diff
@@ -1,6 +1,514 @@
+            let (ret, pair) = parse_der_sequence_defined!(i, parse_der_oid >> parse_der_oid)?;
+            Ok((ret, pair))
+        }
+        let (ret, pairs) = parse_der_sequence_of!(i, parse_oid_pair)?;
+        let mut mappings: HashMap<Oid, Vec<Oid>> = HashMap::new();
+        for pair in pairs.as_sequence().map_err(|e| nom::Err::Failure(e))?.into_iter() {
+            let pair = pair.as_sequence().map_err(|e| nom::Err::Failure(e))?;
+            let left = pair[0].as_oid_val().map_err(|e| nom::Err::Failure(e))?;
+            let right = pair[1].as_oid_val().map_err(|e| nom::Err::Failure(e))?;
+            if left.asn1() == oid!(raw 2.5.29.32.0) || right.asn1() == oid!(raw 2.5.29.32.0) {
+                // mapping to or from anyPolicy is not allowed
+                return Err(Err::Failure(BerError::InvalidTag));
+            }
+            mappings.entry(left)
+                .and_modify(|v| v.push(right.clone()))
+                .or_insert_with(|| vec![right.clone()]);
+        }
+        Ok((ret, PolicyMappings { mappings }))
+    }
+
+    fn parse_inhibitanyplicy<'a>(i: &'a [u8]) -> IResult<&'a [u8], InhibitAnyPolicy, BerError> {
+        let (ret, skip_certs) = map_res!(i, parse_der_integer, |x: BerObject| x.as_u32())?;
+        Ok((ret, InhibitAnyPolicy { skip_certs }))
+    }
+
+    fn parse_extendedkeyusage<'a>(i: &'a [u8]) -> IResult<&'a [u8], ExtendedKeyUsage<'a>, BerError> {
+        let (ret, seq) = parse_der_sequence_of!(i, parse_der_oid)?;
+        let mut seen = std::collections::HashSet::new();
+        let mut eku = ExtendedKeyUsage {
+            any: false,
+            server_auth: false,
+            client_auth: false,
+            code_signing: false,
+            email_protection: false,
+            time_stamping: false,
+            ocscp_signing: false,
+            other: Vec::new(),
+        };
+        for oid in seq.as_sequence().map_err(|e| nom::Err::Failure(e))?.into_iter() {
+            let oid = oid.as_oid_val().map_err(|e| nom::Err::Failure(e))?;
+            if !seen.insert(oid.clone()) {
+                continue;
+            }
+            let asn1 = oid.asn1();
+            if asn1 == oid!(raw 2.5.29.37.0) {
+                eku.any = true;
+            } else if asn1 == oid!(raw 1.3.6.1.5.5.7.3.1) {
+                eku.server_auth = true;
+            } else if asn1 == oid!(raw 1.3.6.1.5.5.7.3.2) {
+                eku.client_auth = true;
+            } else if asn1 == oid!(raw 1.3.6.1.5.5.7.3.3) {
+                eku.code_signing = true;
+            } else if asn1 == oid!(raw 1.3.6.1.5.5.7.3.4) {
+                eku.email_protection = true;
+            } else if asn1 == oid!(raw 1.3.6.1.5.5.7.3.8) {
+                eku.time_stamping = true;
+            } else if asn1 == oid!(raw 1.3.6.1.5.5.7.3.9) {
+                eku.ocscp_signing = true;
+            } else {
+                eku.other.push(oid);
+            }
```

Comment:
Same as earlier: using a `const ` declaration + a `match oid.bytes()` would be better